ethereum/contracts/Broker.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+import "./Verifier.sol";
+import "./Decoder.sol";
+import "./Application.sol";
+
+contract Broker {
+    using Decoder for bytes;
+
+    mapping(address => bool) public applications;
+    Verifier public verifier;
+
+    constructor(address verifierAddr) public {
+        verifier = new Verifier(verifierAddr);
+    }
+
+    /**
+     * @dev registers new applications
+     * @param _appID address _appID is the contract address to be registered as a supported application
+     */
+    function registerApp(address _appID)
```

Comment:
We shouldnt need this, since the broker should hardcode all the applications its knows about. Having a public `registerApp` function also implies that there is some trusted authority who is able to register apps. Our design at least tries to be trustless.

For example, on the Substrate side, the broker uses this hardcoded list of applications: https://github.com/Snowfork/polkadot-ethereum/blob/main/parachain/primitives/core/src/registry.rs

Reply:
https://github.com/Snowfork/polkadot-ethereum/pull/75/commits/b7b0e9cd067a333b960ddca54ef0b5c88e1967c6